pytorch_lightning/loggers/neptune.py

```diff
@@ -273,44 +273,53 @@ def __init__(
         super().__init__()
         self._log_model_checkpoints = log_model_checkpoints
         self._prefix = prefix
+        self._run_name = name
+        self._project_name = project
+        self._api_key = api_key
+        self._run_instance = run
+        self._neptune_run_kwargs = neptune_run_kwargs
+        self._run_short_id = None
+
+        if self._run_instance is not None:
+            self._retrieve_run_data()
 
-        self._run_instance = self._init_run_instance(api_key, project, name, run, neptune_run_kwargs)
+            # make sure that we've log integration version for outside `Run` instances
+            self._run_instance[_INTEGRATION_VERSION_KEY] = __version__
 
-        self._run_short_id = self.run._short_id  # skipcq: PYL-W0212
+    def _retrieve_run_data(self):
         try:
-            self.run.wait()
+            self._run_instance.wait()
+            self._run_short_id = self.run._short_id  # skipcq: PYL-W0212
             self._run_name = self._run_instance["sys/name"].fetch()
         except NeptuneOfflineModeFetchException:
             self._run_name = "offline-name"
 
-    def _init_run_instance(self, api_key, project, name, run, neptune_run_kwargs) -> Run:
-        if run is not None:
-            run_instance = run
-        else:
-            try:
-                run_instance = neptune.init(
-                    project=project,
-                    api_token=api_key,
-                    name=name,
-                    **neptune_run_kwargs,
-                )
-            except NeptuneLegacyProjectException as e:
-                raise TypeError(
-                    f"""Project {project} has not been migrated to the new structure.
-                    You can still integrate it with the Neptune logger using legacy Python API
-                    available as part of neptune-contrib package:
-                      - https://docs-legacy.neptune.ai/integrations/pytorch_lightning.html\n
-                    """
-                ) from e
-
-        # make sure that we've log integration version for both newly created and outside `Run` instances
-        run_instance[_INTEGRATION_VERSION_KEY] = __version__
-
-        # keep api_key and project, they will be required when resuming Run for pickled logger
-        self._api_key = api_key
-        self._project_name = run_instance._project_name  # skipcq: PYL-W0212
+    @property
+    def _neptune_init_args(self):
+        args = {}
+        # Backward compatibility in case of previous version retrieval
```

Comment:
can you put the exact version here?

Reply:
in case it is 0.9, you can use `_NEPTUNE_GREATER_EQUAL_0_9`

Reply:
It's not due to Neptune incompatibility. It's only about pickle-unpickle from previous pytorch-lightning version :wink:

Reply:
Actually we only need a "run id" to recover an already pickled Neptune run + maybe project info + api_token and all that data are stored in previously pickled (for instance pickled with current PTL master) states. Now we need to store a little bit more data that is used on the initialization step of Neptune Run but it's still possible to reuse old pickled loggers and reuse old Neptune runs.

It's only for the purpose of beeing consider as bugfix instead of behavioral change 😎

Reply:
oh! I see. thanks for clarification :)